include/vcpkg/build.h

```diff
@@ -156,6 +156,7 @@ namespace vcpkg
         Triplet triplet;
         bool load_vcvars_env = false;
         bool disable_compiler_tracking = false;
+        bool target_is_xbox = false;
```

Comment:
What about ybox and zbox and aabox? I think vcpkg should have a solution which scales

Reply:
"is xbox" handles the scenario of the Xbox device family in general which are all built at "Windows" platform devices. For the specific device, this is handled through ``XBOX_CONSOLE_TARGET``.

Reply:
The basic design here is that `VCPKG_TARGET_IS_XBOX` indicates that this is an "Xbox" platform build. This causes vcpkg to select the 'xbox.cmake' instead of 'windows.cmake' toolchain -AND- is used to set the **xbox** supports expression value.

The specifics of Xbox variants is handled by the ``XBOX_CONSOLE_TARGET`` string passed via ``VCPKG_CMAKE_CONFIGURE_OPTIONS``. Given that console ecosystems are deprecated or supported via backcompat by design, there's really only ONE 'active' xbox scenario anyhow.